File: output/gpt_4o/avatar/direct_translation/java/python/successful_translation/atcoder_ABC123_C.py

import math

def main():
    N = int(input())
    min_value = 0
    for i in range(5):
        if i == 0:
            min_value = int(input())
        else:
            min_value = min(min_value, int(input()))
    print(int(math.ceil(N / min_value) + 4))

if __name__ == "__main__":
    main()

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/atcoder_ABC123_C.java

import java.util.Scanner;
import java.lang.Math;

public class atcoder_ABC123_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int[] A = new int[5];
        for (int i = 0; i < 5; i++) {
            A[i] = scanner.nextInt();
        }
        int minValue = A[0];
        for (int i = 1; i < 5; i++) {
            if (A[i] < minValue) {
                minValue = A[i];
            }
        }
        int result = (int) Math.ceil((double) N / minValue) + 4;
        System.out.println(result);
    }
}

